internal/reconciler/commonstatus/checker.go

```diff
@@ -0,0 +1,93 @@
+package commonstatus
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kyma-project/telemetry-manager/internal/conditions"
+	"github.com/kyma-project/telemetry-manager/internal/workloadstatus"
+)
+
+const (
+	SignalTypeTraces  = "traces"
+	SignalTypeMetrics = "metrics"
+	SignalTypeLogs    = "logs"
+)
+
+type DeploymentProber interface {
```

Comment:
with this interface introduced at the `common` level, do we still need the same interface in every pipeline?